### PR TITLE
Add Maven Central publication

### PR DESCRIPTION
**Pull request by mohsenoid**
**What**
- Adding Maven Central publication

**Why**
- ASDK-1112

### CODE REVIEW

File: constants.gradle [hunk 1]
```diff
@@ -13,7 +13,7 @@
 // limitations under the License.
 project.ext {
     // ExoPlayer version and version code.
-    releaseVersion = '2.16.1'
+    releaseVersion = '2.16.1.6'
```
Comment (mohsenoid) [inline constants.gradle:16]: ℹ️ We cannot trick the version the same as we did for Jitpack and we need to add our `.x` number in the end.

File: publish.gradle [hunk 2]
```diff
@@ -13,23 +13,44 @@
...
         repositories {
             maven {
+                name = 'mavenRepo'
                 url = findProperty('mavenRepo') ?: "${buildDir}/repo"
             }
+            maven {
+                def releasesRepoUrl = "https://s01.oss.sonatype.org/service/local/staging/deploy/maven2/"
+                def snapshotsRepoUrl = "https://s01.oss.sonatype.org/content/repositories/snapshots/"
+
+                url = releaseVersion.endsWith('SNAPSHOT') ? snapshotsRepoUrl : releasesRepoUrl
```
Comment (mohsenoid) [inline publish.gradle:30]: ℹ️ Adding SNAPSHOT to the end of the version will switch the publication repo URL to maven snapshot.

File: publish.gradle [hunk 3]
```diff
@@ -13,23 +13,44 @@
...
+
+                credentials {
+                    username = System.getenv("OSSRH_USERNAME")
+                    password = System.getenv("OSSRH_PASSWORD")
```
Comment (mohsenoid) [inline publish.gradle:34]: ℹ️ All credentials and keys are added to the GitHub actions secrets. Later we should define a publishing flow.

File: publish.gradle [hunk 4]
```diff
@@ -39,19 +60,32 @@ afterEvaluate {
...
+        def signingSecretKey = System.getenv("GPG_SIGNING_SECRET_KEY")
+        def signingPassword = System.getenv("GPG_SIGNING_PASSWORD")
+        useInMemoryPgpKeys(signingKeyId, signingSecretKey, signingPassword)
+
+        sign publishing.publications.release
+    }
 }
-tasks.withType(PublishToMavenRepository) { it.dependsOn lint, test }
+
+// tasks.withType(PublishToMavenRepository) { it.dependsOn lint, test }
```
Comment (mohsenoid) [inline publish.gradle:88]: ℹ️ I don't get why, but tests are not passing and this line blocks the publishing task.

File: publish.gradle [hunk 5]
```diff
@@ -13,23 +13,44 @@
...
 
 apply plugin: 'maven-publish'
+apply plugin: 'signing'
+
 afterEvaluate {
+
     publishing {
         repositories {
             maven {
+                name = 'mavenRepo'
```
Comment (mohsenoid) [inline publish.gradle:23]: ℹ️ I kept it as mavenRepo cause I have no idea if we want it in the future or not.
Anyhow, we can get rid of it if there are no use cases.

**Review by moyar1532 — APPROVED** [review]

**Pull request merged**